### DISCUSSION TITLE
Gyrotropic Saturated Dipole (Linearized Landau-Lifshitz-Gilbert) Model

### DISCUSSION THREAD

**Discussion by MuhammadShamaas1997** [NanoComp/meep#2213] [General]
Hello,
I want to know about the derivation of Gyrotropic Saturated Dipole (Linearized Landau-Lifshitz-Gilbert) Model (https://meep.readthedocs.io/en/latest/Materials/#gyrotropic-media). Where can I get more information about it's derivation?
How was it derived (linearlised) from the LLG equation, and what are the contributions taken for the effective field (applied, anisotropy, exchange, magnetostatics, eddy current .. etc.?).  Please clarify what Gamma is in this equation (in the LLG equation it should be the gyromagnetic ratio, not a damping factor).  Also clarify how the susceptibility tensor was obtained.

Was the exchange contribution taken into account? I assume since the material is saturated here, so how are spin-waves produced?  If it refers to magnetostatic waves, how are magnetostatic fields calculated? Is M computed directly?

What parameters values were used to determine the susceptibility? What was the assumed resonance mechanism and effective field contributions?. What are the suitable micromagnetic boundary conditions used for the ferrite structure?

How are spin-waves formed when the model enforces a particular permeability spectrum (without actually computing the magnetisation) and does not consider exchange nor even evaluate magnetostatic (demagnetising fields). 

Where can I get material and numerical parameters to carry out some simulations?

Also, please explain about Hz source. The source is not clear: is it a z-directed magnetic current in the transmission line or is it a z-directed plane wave?    

Thank you,
Muhammad Shamaas

**Comment by stevengj** [comment]
cc @seewhydee who implemented this feature for #60 in #863.  @acerjan has also worked on this.

**Comment by stevengj** [comment]
(An Hz source is a z-directed magnetic current.)

**Comment by stevengj** [comment]
(Please stop posting duplicates of your original question.   I've had to delete your duplicates twice now.)

**Comment by seewhydee** [comment]
The equation of motion is obtained by taking the LLG equation, and Taylor expanding in the limit dm << M. That's basically it. There's no additional internal dynamics, such as nonlocal effects.

If more complicated magnetization dynamics is needed, it might be possible to "cheat" by using a material function and updating the parameters of the equation of motion on the fly, but I don't know how feasible that is.

**Reply by MuhammadShamaas1997** [reply]
Thank you for your response. Kindly explain a bit more about the other questions as well : Was the magnetic exchange energy, dipole -dipole energy, anisotropy and Zeeman energy contribution taken into account? How are magnetostatic fields and M calculated? How are spin-waves formed when the model does not consider exchange nor even evaluate magnetostatic (demagnetising fields)?
Thank you,
Muhammad Shamaas

**Reply by MuhammadShamaas1997** [reply]
Please help me.

**Reply by seewhydee** [reply]
Magnetic exchange energy, dipole-dipole energy, etc., all lie outside the assumptions in the magnetization model. The model uses a local LLG equation: the magnetization at a given position, $\mathbf{M}(r)$, obeys an equation of motion that involves $\mathbf{M}(r)$, the local magnetic field $\mathbf{B}(r)$, and nothing else. The equation of motion is exactly as stated in the [docs](https://meep.readthedocs.io/en/latest/Materials/#gyrotropic-media).

In particular, the equation of motion contains no spatial derivatives of $\mathbf{M}(r)$. Such spatial derivatives would arise in the continuum limit from the Heisenberg interaction $-J\mathbf{M}[i]\cdot\mathbf{M}[i+1]$ where $\mathbf{M}[i]$ and $\mathbf{M}[i+1]$ are neighboring spins on a lattice. Such an interaction would give rise to a nonlocal susceptibility, and ultimately dynamical spin waves. There is currently no direct support for that, so there should be no dynamical spin waves in this model. As mentioned, if you really really want to include such dynamics in your electromagnetics simulaations, you might possibly be able to "cheat" by manually updating the susceptibility parameters on each time step.

**Reply by MuhammadShamaas1997** [reply]
Thank you, Sir @seewhydee. What is sigma in linearized LLG model? How is it related to effective magnetic field strength Heff?

**Reply by seewhydee** [reply]
Sorry, I'm not sure what you mean by sigma. You mean conductivity? In Meep, the LLG model is determined in terms of the equation of motion of the polarization, i.e.,

$$\frac{d\mathbf{P}_n}{dt} = \mathbf{b}_n \times \left( - \sigma_n \mathbf{E} + \omega_n \mathbf{P}_n + \alpha_n \frac{d\mathbf{P}_n}{dt} \right) - \gamma_n \mathbf{P}_n$$

(This can be electric polarization or magnetic polarization depending on what field component you attach the susceptibility to.)  How to translate this polarization into a conductivity (whether electric conductivity or magnetic conductivity) is discussed in the [Meep docs](https://meep.readthedocs.io/en/latest/Materials/#conductivity-and-complex). Basically, you identify $J = \partial P/ \partial t$, go to frequency domain by $\partial/\partial t \rightarrow - i \omega$, and define the conductivity as $J = \sigma E$ (for the electric case). For the LLG model, this yields a tensor.

As for effective magnetic field strength, I am not sure how you define that quantity. Again, you have to translate it into magnetic polarization (which is what the MEEP model is expressed in terms of).

**Reply by MuhammadShamaas1997** [reply]
Thank you for the reply Sir @seewhydee. What is sigma_n (σn) in this linearized LLG model equation (equation of motion of polarization)? 

In the documentation (https://meep.readthedocs.io/en/latest/Materials/#material-dispersion) it is written:

"The σn(x) is a user-specified function of position giving the strength of the n-th resonance."

"Note that this equation of motion is completely different from the Drude-Lorentz equation, though the constants σn, ωn, and γn play analogous roles (σn couples the polarization to the driving field, ωn is the angular frequency of precession, and γn is a damping factor)."

What is sigma_n (σn)? Can you tell me about the reference for derivation of this equation: linearized LLG model equation (equation of motion of polarization) so that I can study it in detail?

**Reply by seewhydee** [reply]
It's simply the quantity that connects P to E (or M to H). You can think about it as the background susceptibility.

Maybe the fastest way to proceed is for you to state what you think the equation should be instead, then we can see if Meep's LLG model is compatible or not...
